app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/ApkfyBottomSheetContent.kt

```diff
@@ -101,42 +103,58 @@ class ApkfyBottomSheetContent(private val apkfyState: ApkfyUiState) : BottomShee
 }
 
 @Composable
-fun InfoText(
-  shouldHaveSeeMore: Boolean
-) {
-  var isTextVisible by remember { mutableStateOf(!shouldHaveSeeMore) }
-  if (shouldHaveSeeMore) {
-    Row(
-      modifier = Modifier
-        .fillMaxWidth()
-        .clickable { isTextVisible = !isTextVisible },
-      horizontalArrangement = Arrangement.SpaceBetween
-    ) {
-      Row {
-        Image(
-          modifier = Modifier.size(16.dp),
-          imageVector = getInfo(Palette.GreyLight),
-          contentDescription = null
-        )
-        Text(
-          modifier = Modifier.padding(start = 8.dp),
-          text = stringResource(R.string.see_more_button),
-          style = AGTypography.InputsS,
-          color = Palette.GreyLight
-        )
-      }
+fun ColumnScope.InfoTextC(app: App) {
+  val state = rememberDownloadPermissionState(app)
+  val isTextVisible by remember(state) {
+    derivedStateOf {
+      state is DownloadPermissionState.Allowed || state is DownloadPermissionState.RationaleRejected
```

Comment:
Maybe `SystemNotAllowed` should be considered as well. On sdk versions <= 10, if the external storage permission needs to be requested, it will first be requested without a rationale and if the user denies the system dialog the text won't appear.

Reply:
The requirements in the ticket clearly state, that this text should become visible only if user rejects our own rationale dialog or when download starts (all permissions allowed).
Otherwise there would be no need to differentiate users cancellation of system and rationale dialogs ;-)